### PR TITLE
Add smaller boost to exact match of IDs with no separators

### PR DESCRIPTION
**Pull request by joelverhagen**
Before this change, the exact match mega boost (`^1000`) was only applied for package IDs with dots, hyphens, or underscores in them. The goal of the mega boost was to put make sure a search for `foo.bar.baz` put a package with the matching ID at the top. However we only scoped to boost to package IDs with separators. This scoping was to avoid very common searches like `entity` which should really show [Microsoft.EntityFrameworkCore](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore) at the top not the [Entity](https://www.nuget.org/packages/Entity) package which has very low downloads. More context is here: https://github.com/NuGet/NuGet.Services.Metadata/pull/652.

With the switch to the new SDK, BM25 is now being using on all new indexes. [This is mandated by Azure Search](https://docs.microsoft.com/en-us/azure/search/index-ranking-similarity). This similarity algorithm change results in slightly different results from the Classic similarity which is in PROD right now. From the `SearchScorer` tool, the only notable regressions were related to exact match queries for IDs without separators (i.e. the case that was excluded until now).

This PR adds a smaller boost for exact match queries on package IDs _without_ separators. Based on testing, a boost of (`^10`) is the sweet spot. `entity` shows Microsoft.EntityFrameworkCore at the top and `sqlite` shows `SQLite` at the top because the relative download counts on the exact match packages are so different.

By the way, the long term fix is to highlight the exact match in the search UI clearly and just put all exact matches at the top no matter what. This is tracked here: https://github.com/NuGet/NuGetGallery/issues/7463.

The goal of this PR is to make a targeted fix in the relevancy algorithm to unblock the index rebuild needed for Azure Search Managed Identities.

The search scorer is showing this (**higher on the treatment is better**):

```
Curated Search Queries
======================
Control:   0.740516068252731
Treatment: 0.741433678994227

Client Curated Search Queries
=============================
Control:   0.673364861619194
Treatment: 0.675891116995111

Feedback
========
Control:   0.701412154740343
Treatment: 0.702295157305885
```

In other words, there's not much of a change. If there's any change at all, we expect a minor improvement.

### CODE REVIEW

**Review by loic-sharma — APPROVED** [review]

**Comment by shishirx34** [comment]
Nevermind, I see the discussion on the thread in the linked issue.

**Comment by joelverhagen** [comment]
> I wish we don't have to play with these tweaks and show exact ID match at the top always (with appropriate label). Any thoughts on how trivial/non-trivial such a change can be? Is the only blocker here a UX review?
> 
> This PR looks good though. :shipit:

The gallery change would be pretty easy. Just enhance the UI to show an exact match color/label/border if the package ID == the search query. It is trivial.

For VS PM UI ((90%+ of total searches) we'd need to ship a change in VS. This is more time consuming and would take time to reach the majority of our users.
